lenskit-core/src/main/java/org/grouplens/lenskit/data/history/History.java

```diff
@@ -75,6 +75,8 @@ private History() {}
      * @return An empty history for the user.
      */
     @Nonnull
+    @SafeVarargs
+    @SuppressWarnings("varargs") // method is safe MDE 2015-05-08
     public static <E extends Event> UserHistory<E> forUser(long id, E... events) {
         List<E> list = ImmutableList.copyOf(events);
```

Comment:
I've gone and asked Stack Overflow about this. http://stackoverflow.com/q/30130262/1385039